modules/xmpp/moderator.js

```diff
@@ -314,10 +314,19 @@ export default class Moderator {
                 // decide to send our IQ.
                 this.connection.flush();
             } else {
+                const conferenceRequest = this._createConferenceRequest(roomJid);
+
+                if (FeatureFlags.isJoinAsVisitorSupported()) {
+                    if (!conferenceRequest.properties) {
```

Comment:
I don't think that's necessary. It's always initialized as an object in `_createConferenceRequest`

Reply:
Also, why do we include `visitors-version` only over HTTP? I'd move this to `_createConferenceRequest`